### DISCUSSION TITLE
cpu爆满。之前执行不会的。

### DISCUSSION THREAD

**Discussion by qianxiaoaa** [kkkgo/PaoPaoDNS#66] [问答和帮助]
如下命令
docker run -d \
--name paopaodns \
-v /www/mydata:/data \
-e UPDATE=weekly \
-e CNAUTO=yes \
-e CNFALL=no \
-e CN_TRACKER=yes \
-e IPV6=only6 \
-e CUSTOM_FORWARD=172.17.0.1:53  \
-e  AUTO_FORWARD=yes   \
-e  AUTO_FORWARD_CHECK=yes \
-e  USE_MARK_DATA=yes \
-e  CN_TRACKER=yes \
-e  ADDINFO=yes \
-e  SHUFFLE=yes \
-e  RULES_TTL=604800 \
-e  TZ=Asia/Shanghai \
--restart unless-stopped \
-p 53:53/tcp -p 53:53/udp \
-p 5304:5304/tcp -p 5304:5304/udp \
sliamb/paopaodns

**Comment by kkkgo** [comment]
贴出容器内执行debug.sh和docker logs paopaodns

**Reply by qianxiaoaa** [reply]
好的 骚等

**Reply by qianxiaoaa** [reply]
我是服务器的，我在重装试试

**Reply by qianxiaoaa** [reply]
不好意思 刚才有电脑  上面是docker logs paopaodns的  debug.sh等会

**Comment by qianxiaoaa** [comment]
不知道为啥CPU一直爆满8小时还不降下来

**Reply by kkkgo** [reply]
执行top命令看是哪个程序占用CPU。

**Reply by qianxiaoaa** [reply]
mosdns

**Reply by qianxiaoaa** [reply]
是在容器运行的

**Reply by kkkgo** [reply]
是容器一启动后就占用CPU高吗？
不设置任何环境变量的情况下，默认参数启动是否占用CPU高？

**Reply by qianxiaoaa** [reply]
不是，默认的就不会

**Comment by kkkgo** [comment]
如果默认的环境变量不会导致CPU高占用，可以尝试逐项修改环境变量内容看看是哪个选项会导致。
比如：
把ADDINFO、SHUFFLE设置为no。
把CNFALL设置为yes。
不设置CUSTOM_FORWARD。
IPV6设置为no。
USE_MARK_DATA设置为no。
如果排查到了导致高占用的选项，编辑`/data/mosdns.yaml`修改`    level: error`为`    level: debug`，观察容器日志。

**Comment by qianxiaoaa** [comment]
解决是FORWARD 不用设置这个 把这个删除就可以了

**Reply by kkkgo** [reply]
如果你是指设置`CUSTOM_FORWARD`导致CPU占用高，看到你的参数是一个内网地址，建议排查一下你所转发的DNS服务器的上游是否配置成转发回来了导致死循环，你可以先设置一个其他的DNS服务地址，比如`CUSTOM_FORWARD=8.8.8.8:53`，看看转发功能是否正常工作和CPU是否正常。

**Reply by qianxiaoaa** [reply]
不设置后cpu是不饱满了 但是负载一直很高 
docker run -d \
--name paopaodns \
-v /www/mydata:/data \
-e UPDATE=weekly \
-e CNAUTO=yes \
-e CNFALL=yes \
-e CN_TRACKER=yes \
-e IPV6=yes \
-e  AUTO_FORWARD=yes   \
-e  AUTO_FORWARD_CHECK=yes \
-e  USE_MARK_DATA=yes \
-e SHUFFLE=yes \
-e  CN_TRACKER=yes \
-e  ADDINFO=yes \
-e  SHUFFLE=yes \
-e  RULES_TTL=604800 \
-e  TZ=Asia/Shanghai \
--restart unless-stopped \
-p 53:53/tcp -p 53:53/udp \
-p 5304:5304/tcp -p 5304:5304/udp \
-p 5303:5303/tcp -p 5303:5303/udp \
-p 5302:5302/tcp -p 5302:5302/udp \
-p 5301:5301/tcp -p 5301:5301/udp \
-p 7890:7890/tcp -p 7890:7890/udp \
sliamb/paopaodns

**Reply by kkkgo** [reply]
排查下配置的其他选项。比如配置错误的RULES_TTL可能会导致死循环。
